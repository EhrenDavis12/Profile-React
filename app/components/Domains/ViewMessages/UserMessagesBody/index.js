/**
 *
 * UserMessagesBody
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';
// import styled from 'styled-components';
// import BasicButton from 'components/Domains/SharedKernel/Buttons/BasicButton';
import MessageDetails from 'components/Domains/ViewMessages/MessageDetails';

function UserMessagesBody({
  // responseTest,
  userData,
  userMessages,
  selectUserMessage,
  // selectedUserMessage,
}) {
  // const { responseTest, userData, selectUser } = props;
  /* const dataList = userMessages.map(
    x =>
      selectedUserMessage && x.uuid === selectedUserMessage.uuid ? (
        <MessageDetails key={x.uuid} selectedUserMessage={x} />
      ) : (
        <BasicButton key={x.uuid} func={selectUserMessage} obj={x}>
          subject: {x.subject}
        </BasicButton>
      ),
  ); */
  let dataList;
  if (userMessages) {
    dataList = userMessages.map(x => (
      <MessageDetails
        key={x.uuid}
        userMessage={x}
        selectUserMessage={selectUserMessage}
      />
    ));
  }

  return (
    <div>
      <div>Your google ID is: {userData.auth0Id}</div>
      {dataList}
    </div>
  );
}

UserMessagesBody.propTypes = {
  // responseTest: PropTypes.arrayOf(PropTypes.shape({})),
  userMessages: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  // selectedUserMessage: PropTypes.shape({}),
  userData: PropTypes.shape({
    auth0Id: PropTypes.string.isRequired,
  }).isRequired,

  selectUserMessage: PropTypes.func.isRequired,
};

export default UserMessagesBody;
