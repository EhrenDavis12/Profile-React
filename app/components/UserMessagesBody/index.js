/**
 *
 * UserMessagesBody
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';
import BasicButton from 'components/SharedKernel/Buttons/BasicButton';

function UserMessagesBody({
  // responseTest,
  response,
  userMessages,
  selectUserMessage,
}) {
  // const { responseTest, response, selectUser } = props;
  const dataList = userMessages.map(x => (
    <BasicButton key={x.uuid} func={selectUserMessage} obj={x}>
      subject: {x.subject}
    </BasicButton>
  ));
  return (
    <div className="container">
      <div>Your google ID is: {response.data.auth0Id}</div>
      {dataList}
    </div>
  );
}

UserMessagesBody.propTypes = {
  // responseTest: PropTypes.arrayOf(PropTypes.shape({})),
  userMessages: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  response: PropTypes.shape({
    data: PropTypes.shape({
      auth0Id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,

  selectUserMessage: PropTypes.func.isRequired,
};

export default UserMessagesBody;
