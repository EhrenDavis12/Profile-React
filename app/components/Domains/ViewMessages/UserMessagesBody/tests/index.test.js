import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

import UserMessagesBody from '../index';

const userMessage = {
  subject: 'subject string',
  createdAt: `01-01-2000 T 01:01:1000`,
  message: 'This is a test message',
};
const userMessages = [
  { ...userMessage, uuid: 'key1' },
  { ...userMessage, uuid: 'key2' },
];
const userData = { auth0Id: 'auth0Id' };

describe('<UserMessagesBody /> snap Shots', () => {
  it('renders as expected with populated messages', () => {
    const mockCallBack = jest.fn(() => true);
    const tree = renderer.create(
      <UserMessagesBody
        selectUserMessage={mockCallBack}
        userMessages={userMessages}
        userData={userData}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders as expected with null messages', () => {
    const mockCallBack = jest.fn(() => true);
    const tree = renderer.create(
      <UserMessagesBody
        selectUserMessage={mockCallBack}
        userMessages={null}
        userData={userData}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('Evaluate Children counts', () => {
  it('Mount Should have array of children', () => {
    const mockCallBack = jest.fn(() => true);
    const wrapper = mount(
      <UserMessagesBody
        selectUserMessage={mockCallBack}
        userMessages={userMessages}
        userData={userData}
      />,
    );
    expect(wrapper.find('MessageDetails').length).toEqual(2);
  });

  it('Mount Should have array of children', () => {
    const mockCallBack = jest.fn(() => true);
    const wrapper = mount(
      <UserMessagesBody
        selectUserMessage={mockCallBack}
        userMessages={null}
        userData={userData}
      />,
    );
    expect(wrapper.find('MessageDetails').length).toEqual(0);
  });
});
