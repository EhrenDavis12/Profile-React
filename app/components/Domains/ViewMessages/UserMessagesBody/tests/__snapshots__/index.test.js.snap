// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMessagesBody /> snap Shots renders as expected with null messages 1`] = `
<UserMessagesBody
  selectUserMessage={[MockFunction]}
  userData={
    Object {
      "auth0Id": "auth0Id",
    }
  }
  userMessages={null}
>
  <div
    className="MessageDetails-Box"
  >
    <div>
      Your google ID is: 
      auth0Id
    </div>
  </div>
</UserMessagesBody>
`;

exports[`<UserMessagesBody /> snap Shots renders as expected with populated messages 1`] = `
<UserMessagesBody
  selectUserMessage={[MockFunction]}
  userData={
    Object {
      "auth0Id": "auth0Id",
    }
  }
  userMessages={
    Array [
      Object {
        "createdAt": "01-01-2000 T 01:01:1000",
        "message": "This is a test message",
        "subject": "subject string",
        "uuid": "key1",
      },
      Object {
        "createdAt": "01-01-2000 T 01:01:1000",
        "message": "This is a test message",
        "subject": "subject string",
        "uuid": "key2",
      },
    ]
  }
>
  <div
    className="MessageDetails-Box"
  >
    <div>
      Your google ID is: 
      auth0Id
    </div>
    <MessageDetails
      key="key1"
      selectUserMessage={[MockFunction]}
      userMessage={
        Object {
          "createdAt": "01-01-2000 T 01:01:1000",
          "message": "This is a test message",
          "subject": "subject string",
          "uuid": "key1",
        }
      }
    >
      <div
        className="MessageDetails-Box"
      >
        <BasicButton
          obj={
            Object {
              "createdAt": "01-01-2000 T 01:01:1000",
              "message": "This is a test message",
              "subject": "subject string",
              "uuid": "key1",
            }
          }
          onClick={[MockFunction]}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="subject"
              key=".0"
            >
              subject: 
              subject string
            </div>
            <div
              className="date"
              key=".1"
            >
              Date: 
              01-01-2000 
            </div>
            <div
              className="time"
              key=".2"
            >
              Time: 
               01:01:1000
            </div>
            <div
              className="Message wordWrap"
              key=".3"
            >
              Message: 
              This is a test message
            </div>
          </div>
        </BasicButton>
      </div>
    </MessageDetails>
    <MessageDetails
      key="key2"
      selectUserMessage={[MockFunction]}
      userMessage={
        Object {
          "createdAt": "01-01-2000 T 01:01:1000",
          "message": "This is a test message",
          "subject": "subject string",
          "uuid": "key2",
        }
      }
    >
      <div
        className="MessageDetails-Box"
      >
        <BasicButton
          obj={
            Object {
              "createdAt": "01-01-2000 T 01:01:1000",
              "message": "This is a test message",
              "subject": "subject string",
              "uuid": "key2",
            }
          }
          onClick={[MockFunction]}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="subject"
              key=".0"
            >
              subject: 
              subject string
            </div>
            <div
              className="date"
              key=".1"
            >
              Date: 
              01-01-2000 
            </div>
            <div
              className="time"
              key=".2"
            >
              Time: 
               01:01:1000
            </div>
            <div
              className="Message wordWrap"
              key=".3"
            >
              Message: 
              This is a test message
            </div>
          </div>
        </BasicButton>
      </div>
    </MessageDetails>
  </div>
</UserMessagesBody>
`;
